src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -500,6 +510,26 @@ void ImageReplayer<I>::handle_init_remote_journaler(int r) {
     return;
   }
 
+  m_remote_journaler->add_listener(&m_remote_listener);
+
+  cls::journal::Client client;
+  r = m_remote_journaler->get_cached_client(m_local_mirror_uuid, &client);
+  if (r < 0) {
+    derr << "error retrieving remote journal client: " << cpp_strerror(r)
+	 << dendl;
+    on_start_fail(r, "error retrieving remote journal client");
+    return;
+  }
+
+  dout(20) << "client=" << client << dendl;
+
+  if (client.state != cls::journal::CLIENT_STATE_CONNECTED) {
```

Comment:
@dillaman  Do you mean "check if it is disconnected" logic? It is here, after add_listener, just to recheck that the client was not disconnected in time window between we checked the state in bootstrap and registered the listener (so we lost the notification). Or do you want the listener is registered in bootstrap?